server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueueTest.java

```diff
@@ -169,23 +170,29 @@ void givenANonDeserializableTaskItShouldBeFlaggedAsFailedAndItDoesNotPreventFoll
 
     @Test
     void tasksShouldBeConsumedSequentially() {
-        Task task1 = new CompletedTask();
+        AtomicLong counter = new AtomicLong(0L);
+
+        Task task1 = new MemoryReferenceTask(() -> {
+            counter.addAndGet(1);
+            Thread.sleep(1000);
+            return Task.Result.COMPLETED;
+        });
         TaskId taskId1 = TaskId.fromString("1111d081-aa30-11e9-bf6c-2d3b9e84aafd");
         TaskWithId taskWithId1 = new TaskWithId(taskId1, task1);
 
-        Task task2 = new CompletedTask();
+        Task task2 =  new MemoryReferenceTask(() -> {
+            counter.addAndGet(2);
+            return Task.Result.COMPLETED;
+        });
+
         TaskId taskId2 = TaskId.fromString("2222d082-aa30-22e9-bf6c-2d3b9e84aafd");
         TaskWithId taskWithId2 = new TaskWithId(taskId2, task2);
 
-        when(worker.executeTask(taskWithId1)).then(answer -> {
-            TimeUnit.MINUTES.sleep(2);
-            return Mono.just(Task.Result.COMPLETED);
-        });
-
         testee.submit(taskWithId1);
         testee.submit(taskWithId2);
 
-        verify(worker, timeout(100)).executeTask(taskWithId1);
-        verifyNoMoreInteractions(worker);
+        assertThatThrownBy(() -> await().atMost(500, TimeUnit.MILLISECONDS).untilAtomic(counter, CoreMatchers.equalTo(3L)))
+            .hasMessageContaining("expected <3L> but was <1L>");
```

Comment:
it looks really brittle, why do you have to match an assertion message for that test?